practice_2/ViewController.swift

```diff
@@ -1,19 +1,93 @@
-//
-//  ViewController.swift
-//  practice_2
-//
-//  Created by user199993 on 6/27/21.
-//
-
 import UIKit
 
 class ViewController: UIViewController {
-
+    var models = [Model]()
+    @IBOutlet weak var tableView: UITableView!
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        tableView.delegate = self
+        tableView.dataSource = self
+        
+        ApiManager.shared.getRandom { images in
+            for link in images.message! {
+                self.models.append(Model(imageLink: link))
+            }
+        }
+        
+        // we use test models if we get https errors or not corrected data
+        if models.count != 6 {
+            createTestModels()
+        }
     }
+    
+    private func createTestModels() {
+        self.models = [Model]()
+        models.append(Model(imageLink: "https://images.dog.ceo/breeds/bulldog-boston/n02096585_318.jpg"))
+        models.append(Model(imageLink: "https://images.dog.ceo/breeds/briard/n02105251_7430.jpg"))
+        models.append(Model(imageLink: "https://images.dog.ceo/breeds/akita/512px-Ainu-Dog.jpg"))
+        models.append(Model(imageLink: "https://images.dog.ceo/breeds/mountain-swiss/n02107574_497.jpg"))
+        models.append(Model(imageLink: "https://images.dog.ceo/breeds/ridgeback-rhodesian/n02087394_7777.jpg"))
+        models.append(Model(imageLink: "https://images.dog.ceo/breeds/setter-irish/n02100877_585.jpg"))
+        models.shuffle()
+    }
+}
 
+extension ViewController: UITableViewDataSource, UITableViewDelegate {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
```

Comment:
table should contain all items in dogs list.
custom cells should repeat.

Reply:
Fixed this point. Hope I understood you correctly.